### PR TITLE
Introduces the single bodysite notebook

### PR DESCRIPTION
**Pull request by JWDebelius**
Creates a notebook for generating the single sample per bodysite packaged results.

### CODE REVIEW

**Comment by JWDebelius** [comment]
I fixed the pep8, but the `parallel_alpha_diversity` is still timing out.

**Comment by mortonjt** [comment]
I noticed the timing out issue with https://github.com/biocore/American-Gut/pull/181

Can you try replacing the last line in [.travis.yml](https://github.com/biocore/American-Gut/blob/master/.travis.yml) with
```
travis_wait ./scripts/run_the_ipys.sh ipynb/primary-processing/
```

This will allocate 20 minutes to run the scripts rather than just 10 minutes before travis crashes.

File: americangut/util.py [hunk 1]
```diff
@@ -525,3 +530,97 @@ def clean_and_reformat_mapping(in_fp, out_fp, body_site_column_name,
...
+        columns include 'HOST_SUBJECT_ID' for the unique individual identifier,
+        and 'DEPTH', containing the sequencing depth.
+    depths: iterable
+        The rarefaction depths used for analysis. Depths may be an iterable
+        of floats or castable strings.
+
+    Returns
+    -------
+    single_ids : dict
+        A list of sample IDs where each ID represents a single sample for a
```
Comment (wasade) [inline americangut/util.py]: Can you reword this? Something like: "A dict keyed by rarefaction depth, and valued by singular sample IDs per subject"

File: americangut/util.py [hunk 2]
```diff
@@ -525,3 +530,97 @@ def clean_and_reformat_mapping(in_fp, out_fp, body_site_column_name,
...
+        by the key.
+
+    """
+    num_depths = len(depths)
+
+    single_ids = {depth: [] for depth in depths}
+    single_ids['unrare'] = []
+
+    for hsi, subject in map_.groupby('HOST_SUBJECT_ID'):
+        for depth_id, depth in enumerate(depths[::-1]):
```
Comment (wasade) [inline americangut/util.py]: this implies the `depths` are sorted coming in. Can the docstring be updated to reflect this, or can an explicit sort be used to ensure it?

File: americangut/util.py [hunk 3]
```diff
@@ -525,3 +530,97 @@ def clean_and_reformat_mapping(in_fp, out_fp, body_site_column_name,
...
+            # Checks the subject has at least one sample at the specified depth
+            if (subject['depth'].astype(float) >= float(depth)).any():
```
Comment (wasade) [inline americangut/util.py]: Why not just do this cast a single time outside of the nested forloop?

File: americangut/util.py [hunk 4]
```diff
@@ -525,3 +530,97 @@ def clean_and_reformat_mapping(in_fp, out_fp, body_site_column_name,
...
+                # Chooses one sample at random
+                id_ = np.random.choice(
+                    subject.loc[subject['depth'] >= float(depth)].index,
+                    replace=False
+                    )
+                single_ids[depth].append(id_)
+                break
+        else:
```
Comment (wasade) [inline americangut/util.py]: Is this doing what you expect? I think it is, but please please document the desired behavior. The for/else is unusual in a programming language and not common to encounter in python despite its support.

File: ipynb/primary-processing/01-get_sequences_and_metadata.md [hunk 5]
```diff
@@ -25,6 +25,7 @@ Now let's actually fetch the study data. `fetch_study` will only pull down acces
 >>> for accession in study_accessions:
 ...     new_samples = agu.fetch_study(accession, chp_path)
 >>> print "New samples received: %d" % new_samples
+New samples received: 0
```
Comment (wasade) [inline ipynb/primary-processing/01-get_sequences_and_metadata.md]: can you please scrub output from the notebooks? it looks like the changes to all of the notebooks were not intentional to include

File: ipynb/primary-processing/02-filter_sequences_for_blooms.md [hunk 6]
```diff
@@ -97,4 +99,32 @@ Finally, let's do a quick sanity check that we have sequence data, and that the
...
+
+Counts/sample detail:
+10317.000002244: 1.0
+10317.000007118: 1.0
+10317.000005844: 59.0
+10317.000002035: 303.0
+10317.000009111: 500.0
+```
+
+```python
```
Comment (wasade) [inline ipynb/primary-processing/02-filter_sequences_for_blooms.md]: for instance, these changes should not be part of the pull request. I'm concerned that you may be doing `git add *.md`, if so, I recommend against that as it risks committing unintended changes.

File: ipynb/primary-processing/03-pick_otus.md [hunk 7]
```diff
@@ -52,4 +52,31 @@ And we'll end with some sanity checking of the outputs.
 ...     print "Trim: %s" % trim
 ...     print '\n'.join(summary[:10])
 ...     print
+Trim: notrim
+Num samples: 15
+Num observations: 11
```
Comment (wasade) [inline ipynb/primary-processing/03-pick_otus.md]: same

File: ipynb/primary-processing/04-prepare_metaanalyses.md [hunk 8]
```diff
@@ -107,19 +107,57 @@ Before we proceed, let's dump out a little information about the two merged tabl
 ```python
 >>> summary = !biom summarize-table -i $ag_gg_100nt_biom
 >>> print '\n'.join(summary[:10])
+Num samples: 25
+Num observations: 6847
+Total count: 19597078
```
Comment (wasade) [inline ipynb/primary-processing/04-prepare_metaanalyses.md]: same, will stop on these comments

File: ipynb/primary-processing/05-alpha_diversity.md [hunk 9]
```diff
@@ -64,16 +65,51 @@ Once we have the rarefactions, we can then compute the diversity of every sample
...
 
 ```python
 >>> for table, depth, rarefaction, multi_directory, adiv_directory, keybase in parameter_iterator():
 ...     # sometimes QIIME leaves this directory, unsure why.
-...     !rmdir $adiv_directory/ALDIV_*
+...     if os.path.exists(os.path.join(adiv_directory, 'ALDIV_*')):
+...         !rmdir $adiv_directory/ALDIV_*
 ...     !collate_alpha.py -i $adiv_directory \
 ...                       -o $adiv_directory
+Traceback (most recent call last):
```
Comment (wasade) [inline ipynb/primary-processing/05-alpha_diversity.md]: are these tracebacks expected?

File: ipynb/primary-processing/06-beta-diversity.md [hunk 10]
```diff
@@ -134,6 +134,26 @@ For one of the figures, the volume of US samples dominates those from other coun
 ...                            --category COUNTRY \
 ...                            --mapping_file $ag_gg_cleaned_md \
 ...                            --output $ag_gg_100nt_10k_ss_bdiv_u
+Traceback (most recent call last):
```
Comment (wasade) [inline ipynb/primary-processing/06-beta-diversity.md]: same

File: setup.py [hunk 11]
```diff
@@ -39,7 +39,7 @@
       scripts=glob('scripts/*py'),
       setup_requires=['numpy >= 1.9.2'],
       install_requires=[
-          'IPython < 4.0.0',
+          'IPython',
```
Comment (wasade) [inline setup.py]: Is ipymd compatible with >= 4.0.0 now? If not, please revert

File: ipynb/primary-processing/11-generate_per_bodysite_results.md [hunk 12]
```diff
@@ -0,0 +1,253 @@
...
+depths = agenv.get_rarefaction_depth()
+depth_names = ['1k', '10k']
+```
+
+We'll start by splitting the raw OTU table and metadata by the `BODY_HABITAT` field, so we have an OTU table for each bodysite.
+
+```python
+for trim in ['notrim', '100nt']:
+    for depth in ['raw', '1k', '10k']:
+        meta = agu.get_existing_path(agenv.paths['package']['split']['%s-map' % trim])
```
Comment (wasade) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: this is an existing path?

File: ipynb/primary-processing/11-generate_per_bodysite_results.md [hunk 13]
```diff
@@ -0,0 +1,253 @@
...
+        otus = agu.get_existing_path(agenv.paths['package']['split']['%s-%s-otu' % (trim, depth)])
+        odir = agu.get_new_path(agenv.paths['package']['split']['%s-%s-dir' % (trim, depth)])
+
+        !mkdir -p $odir
+        !split_otu_table.py -i $otus \
```
Comment (wasade) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: the split tables are already computed in the collapse notebook:

https://github.com/biocore/American-Gut/blob/master/americangut/notebook_environment.py#L643

File: ipynb/primary-processing/11-generate_per_bodysite_results.md [hunk 14]
```diff
@@ -0,0 +1,253 @@
...
+                            -m $meta \
+                            -o $odir \
+                            -f BODY_HABITAT
+```
+
+Next, we'll move the unrarefied tables, split by bodysite, for each of the bodysites we're interested in. For this, we're going to write a quick parameter iterator, that will let us generate the samples we need.
```
Comment (wasade) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: if they need to be written and then moved, why not just write them to the correct location first?

File: ipynb/primary-processing/foo [hunk 15]
```diff
@@ -0,0 +1,3 @@
...MON_NAME	NON_FOOD_ALLERGIES_DRUG_EG_PENICILLIN	NON_FOOD_ALLERGIES_UNSPECIFIED	FLU_VACCINE_DATE	PKU	SUGAR_SWEETENED_DRINK_FREQUENCY	BMI	ENV_BIOME	CENSUS_REGION	DOG_LOCATION	TEETHBRUSHING_FREQUENCY	ENA-SPOT-COUNT	CONTRACEPTIVE	CONSUME_ANIMAL_PRODUCTS_ABX	LIVINGWITH	RELATIONSHIPS_WITH_OTHERS_IN_STUDY	CLINICAL_CONDITION	ENV_MATTER	FED_AS_INFANT	PHYSICAL_SPECIMEN_REMAINING	LAST_TRAVEL	EPILEPSY_OR_SEIZURE_DISORDER	MILK_CHEESE_FREQUENCY	DNA_EXTRACTED	ALCOHOL_TYPES_BEERCIDER	ROOMMATES	HIGH_FAT_RED_MEAT_FREQUENCY	ALTITUDE	SEX	ALLERGIC_TO_TREE_NUTS	ASD	PREGNANT_DUE_DATE	SOFTENER	LUNG_DISEASE	CAT_CONTACT	ALCOHOL_FREQUENCY	POULTRY_FREQUENCY	FROZEN_DESERT_FREQUENCY	BODY_HABITAT	ROOMMATES_IN_STUDY	APPENDIX_REMOVED	DRINKING_WATER_SOURCE	SEAFOOD_FREQUENCY	BMI_CAT	SURVEY_ID	DIET_TYPE	SUBSET_ANTIBIOTIC_HISTORY	RED_MEAT_FREQUENCY	DEPTH	REQUIRED_SAMPLE_INFO_STATUS	LEVEL_OF_EDUCATION	BREASTMILK_FORMULA_ENSURE	ALCOHOL_CONSUMPTION	Description	BarcodeSequence	MEAT_EGGS_FREQUENCY	RACE	KIDNEY_DISEASE	COLLECTION_TIME	VITAMIN_B_SUPPLEMENT_FREQUENCY	LATITUDE	VEGETABLE_FREQUENCY	SUBSET_IBD	ANTIBIOTIC_HISTORY	POOL_FREQUENCY	PREGNANT	ALCOHOL_TYPES_SOUR_BEERS	SIMPLE_BODY_SITE	TITLE_ACRONYM	TITLE_BODY_SITE	HMP_SITE
```
Comment (wasade) [inline ipynb/primary-processing/foo]: should this file be included?

**Comment by wasade** [comment]
Few comments

File: ipynb/primary-processing/01-get_sequences_and_metadata.md (hunk 5 above)
Comment (JWDebelius) [inline ipynb/primary-processing/01-get_sequences_and_metadata.md]: Sorry! I didn't realize it did that. I will fix and pay better attention in the future.

File: ipynb/primary-processing/11-generate_per_bodysite_results.md (hunk 12 above)
Comment (JWDebelius) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: Yep. It's the source for the split.

File: ipynb/primary-processing/11-generate_per_bodysite_results.md (hunk 13 above)
Comment (JWDebelius) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: But, it doesn't have the accompanying filtered metadata. There's an advantage in my mind to having it split the metadata, too. But, I can also go back to the tables from collapsed.

File: ipynb/primary-processing/11-generate_per_bodysite_results.md (hunk 14 above)
Comment (JWDebelius) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: Because I want them named in a specific way that doesn't match what's used in `split_otu_table.py`.

I also like to keep the steps distinct, because I like to be able to go back a step if I make a mistake with the metadata and save something I shouldn't when I'm doing analysis.

File: ipynb/primary-processing/foo (hunk 15 above)
Comment (JWDebelius) [inline ipynb/primary-processing/foo]: I assumed it was part of the master directory I downloaded. I thought it was supposed to be here. Can remove if thats wrong

File: setup.py (hunk 11 above)
Comment (JWDebelius) [inline setup.py]: Ive only ever run ipymd with IPython 4.0

File: ipynb/primary-processing/11-generate_per_bodysite_results.md (hunk 12 above)
Comment (wasade) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: Is it created earlier in the notebook?

File: ipynb/primary-processing/11-generate_per_bodysite_results.md (hunk 13 above)
Comment (wasade) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: Oh, I see. But the metadata can be filtered based on the ids in the
existing tables, or is that awkward?

File: ipynb/primary-processing/foo (hunk 15 above)
Comment (wasade) [inline ipynb/primary-processing/foo]: It isn't part of the repo from what I can see

File: setup.py (hunk 11 above)
Comment (wasade) [inline setup.py]: The editing capabilities were broken when I tried back in October, and I
thought you had indicated so as well? Will test when on my laptop next

File: ipynb/primary-processing/11-generate_per_bodysite_results.md (hunk 12 above)
Comment (JWDebelius) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: It's the metadata.txt file from 01-raw.

File: ipynb/primary-processing/11-generate_per_bodysite_results.md (hunk 13 above)
Comment (JWDebelius) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: Yes, but I find it more awkward. I was trying to replicate your style of using as many QIIME calls as possible. I didn't do a great job, but this was one where it made sense.

**Comment by JWDebelius** [comment]
@mortonjt: I added the line, and now it can't parse the travis.yml file. I thought I'd copied correctly? I also thought Embriette had made this change already, so I'm kind of surprised its running into the error. If my PR doesn't fix this, would it be possible for someone with more experience with travis to address it systematically?

File: setup.py (hunk 11 above)
Comment (JWDebelius) [inline setup.py]: @wasade, did you get a chance to test this?

**Comment by JWDebelius** [comment]
@mortonjt, @squirrelo, @wasade, @EmbrietteH, could I get another review?

Thanks!

File: setup.py (hunk 11 above)
Comment (wasade) [inline setup.py]: ipymd hasn't had a commit in months, so I highly suspect the issue still
exists. When we spoke over the on Friday, you indicated you were doing an
explicit conversion from .ipynb to .md which suggests it is broken

Comment (JWDebelius) [inline setup.py]: Okay, I just tested it. And, you can start with an .md file, if thats what you prefer. I find the explicit conversion from ipynb to markdown easier... 

I think the main difference between 3.0 and 4.0 in getting the conversion working is which config file you modify. In 3.0, I think its and ipython config and in 4.0 its a jupyter config.

Comment (wasade) [inline setup.py]: That means it isn't working: if it is working, you should not notice a difference when operating on a .md or .ipynb as the conversion is implicit

Comment (JWDebelius) [inline setup.py]: This is an issue of personal preference, not functionality. I don't like opening the markdown file, editing it, and then opening the notebook with IPython. I can, and the package works fine. (Actually, it only works that way with IPython 4.0, because there's a backwards compatibility changed between 3.0 and 4.0 in where the file is changed.) But, I'd rather operate in the ipynb framework and then do the conversion.

File: americangut/util.py [hunk 16]
```diff
@@ -525,3 +530,108 @@ def clean_and_reformat_mapping(in_fp, out_fp, body_site_column_name,
...
+    alpha_map_ = alpha_map_.loc[keep]
+
+    return alpha_map_
+
+
+def get_single_id_lists(map_, depths):
+    """Identifies a single sample per individual
+
+    Single samples are identified based on host subject id, and then by
+    randomly selecting samples from avaliable samples at the highest
```
Comment (mortonjt) [inline americangut/util.py]: avaliable -> available

File: americangut/util.py [hunk 17]
```diff
@@ -525,3 +530,108 @@ def clean_and_reformat_mapping(in_fp, out_fp, body_site_column_name,
...
+    rarefaction depth. If an individual has multiple samples, but only one
+    above the rarefaction threshhold, the sample with the higher sequencing
+    depth is selected, and then inherieted across lists. If there are two
```
Comment (mortonjt) [inline americangut/util.py]: inherieted -> inherited

File: americangut/util.py [hunk 18]
```diff
@@ -525,3 +530,108 @@ def clean_and_reformat_mapping(in_fp, out_fp, body_site_column_name,
...
+    return alpha_map_
+
+
+def get_single_id_lists(map_, depths):
+    """Identifies a single sample per individual
+
+    Single samples are identified based on host subject id, and then by
+    randomly selecting samples from avaliable samples at the highest
+    rarefaction depth. If an individual has multiple samples, but only one
+    above the rarefaction threshhold, the sample with the higher sequencing
```
Comment (mortonjt) [inline americangut/util.py]: threshhold -> threshold

File: americangut/util.py [hunk 19]
```diff
@@ -525,3 +530,108 @@ def clean_and_reformat_mapping(in_fp, out_fp, body_site_column_name,
...
+    depth is selected, and then inherieted across lists. If there are two
+    samples above the same threshhold, the sample is selected randomly.
```
Comment (mortonjt) [inline americangut/util.py]: threshhold -> threshold

File: ipynb/primary-processing/11-generate_per_bodysite_results.md [hunk 20]
```diff
@@ -0,0 +1,278 @@
+Analysis focused on a paricular dataset often starts with seperating the data by bodysite. This notebook, and subsequent notebooks (12-15) will seperate the American Gut dataset by bodysite and rarefaction depth.
```
Comment (mortonjt) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: seperating -> separating

File: ipynb/primary-processing/11-generate_per_bodysite_results.md [hunk 21]
```diff
@@ -0,0 +1,278 @@
...
+...                     agu.get_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['sink-%s-weighted-unifrac' % depth]),
+...                     ]
+...
+...                 yield (bodysite, depth, source_otu, source_map, source_alpha, source_beta, sink_dir, sink_otu, sink_map, sink_beta)
+```
+
+Let's use our helper function to add the alpha diversity to our mapping file, and move the rest of the data we need.
+
+```python
+>>> for (bodysite, depth, source_otu, source_map, source_alpha, source_beta, sink_dir, sink_otu, sink_map, sink_beta) in rarefied_parameter_iterator():
```
Comment (mortonjt) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: Is there documentation elsewhere that explains what sinks and sources are?  If not, it may be preferable to have an explanation here.

File: ipynb/primary-processing/11-generate_per_bodysite_results.md [hunk 22]
```diff
@@ -0,0 +1,278 @@
...
+...
+...     # Moves the OTU table
+...     !cp $source_otu $sink_otu
+...
+...     # Moves the beta diversity tables
+...     for source, sink in zip(*(source_beta, sink_beta)):
+...         !cp $source $sink
+```
+
+We're going to select a single sample for each subject at each body site. The human microbiome project [doi: ] demonstrated that differences across bodysites are larger than interpersonal differences within a single bodysite. However, other studies (ie Caporaso 2011 [[doi: 10.1186/gb-2011-12-5-r50](http://www.ncbi.nlm.nih.gov/pubmed/21624126)], Wu 2011 [[doi: 10.1126/science](http://www.ncbi.nlm.nih.gov/pubmed/21885731)]) suggest a high degree of stability within individual communities. A single sample can be use to avoid the bias associated with multiple closely related samples from a single individual.
```
Comment (mortonjt) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: Don't forget the doi for the human microbiome project

File: ipynb/primary-processing/11-generate_per_bodysite_results.md [hunk 23]
```diff
@@ -0,0 +1,278 @@
...
+
+We select the sample using a constrained, random method. Samples which are sequenced at depths greater than the highest rarefaction depth are given priority. However, if multiple samples exist greater than this depth, the sample is selected at random. The process is repeated for each rarefaction depth, until a sample can be identified. This way, the same sample can be retained across analysis performed at multiple depths.
```
Comment (mortonjt) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: random method -> randomized method?

File: ipynb/primary-processing/11-generate_per_bodysite_results.md [hunk 24]
```diff
@@ -0,0 +1,278 @@
...
+
+We're going to define another helper function, to work through the single sample files. We're then going to filter the data.
+
+```python
+>>> def single_sample_iterator():
+...     for bodysite in ['fecal', 'oral', 'skin']:
+...         for depth_name in depth_names:
+...             single_fp = agu.get_existing_path(agenv.paths['package']['single_ids']['%s-%s' % (bodysite, depth_name)])
+...             for trim in ['notrim', '100nt']:
+...                 source_otu = agu.get_existing_path(agenv.paths['package']['all_participants_one_sample'][bodysite][trim]['source-%s-otu' % depth_name])
```
Comment (mortonjt) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: One option that could make this more readable is to retrieve the subdictionary
`agenv.paths['package']['all_participants_one_sample'][bodysite][trim]`, and only use this dictionary for assigning the remaining variables

File: ipynb/primary-processing/11-generate_per_bodysite_results.md [hunk 25]
```diff
@@ -0,0 +1,278 @@
...
+...     smap.to_csv(sink_map, sep='\t', index_label='#SampleID')
+```
+
+Finally, we're going to go through the fecal samples, and generate a set of samples loosely described as "healthy adults". These are individuals 20-69 with a BMI between 18.5 and 30 who report no history of diabetes, inflammatory bowel disease, and have not used antibiotics in the past year.
+
+```python
+>>> for trim in ['notrim', '100nt']:
+...     for sample_number in ['all_samples', 'one_sample']:
+...         for depth_name in ['1k', '10k']:
+...             source_otu = agu.get_existing_path(agenv.paths['package']['sub_participants_%s' % sample_number]['fecal'][trim]['source-%s-otu' % depth_name])
```
Comment (mortonjt) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: Same here

File: ipynb/primary-processing/11-generate_per_bodysite_results.md [hunk 26]
```diff
@@ -0,0 +1,278 @@
...
+
+Finally, we're going to check that we've generated the files we expect.
+
+```python
+>>> error = False
+...
+>>> for path in agenv.paths['package']['split'].values():
+...     filepath = agu.get_path(path)
+...     if not os.path.exists(filepath):
+...             print "Could not find: %s" % filepath
```
Comment (mortonjt) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: This looks like Python 2 syntax ...
Not sure if this is a problem

Comment (JWDebelius) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: Everything else in the repo is Python 2.

File: ipynb/primary-processing/11-generate_per_bodysite_results.md (hunk 23 above)
Comment (JWDebelius) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: It's not a method for randomization, it's a method that utilizes a random number generator. Thus, "random method".

**Comment by JWDebelius** [comment]
Addressed comments. @mortonjt, how does it look?

**Comment by JWDebelius** [comment]
Bumping this because its needed for alpha diversity plots later

File: ipynb/primary-processing/11-generate_per_bodysite_results.md (hunk 21 above)
Comment (mortonjt) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: Don't forget to address this

File: ipynb/primary-processing/11-generate_per_bodysite_results.md (hunk 22 above)
Comment (mortonjt) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: Ping on this

File: ipynb/primary-processing/11-generate_per_bodysite_results.md (hunk 24 above)
Comment (mortonjt) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: Don't forget to address this

File: ipynb/primary-processing/11-generate_per_bodysite_results.md (hunk 21 above)
Comment (JWDebelius) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: That was addressed elsewhere.

Comment (mortonjt) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: Cool.

File: ipynb/primary-processing/11-generate_per_bodysite_results.md (hunk 24 above)
Comment (JWDebelius) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: This has been addressed. Im not sure why you're seeing the outdated version

File: ipynb/primary-processing/11-generate_per_bodysite_results.md (hunk 22 above)
Comment (JWDebelius) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: This is also an outdated change. Again, not sure why Git isn't showing the new version.

File: ipynb/primary-processing/11-generate_per_bodysite_results.md (hunk 24 above)
Comment (mortonjt) [inline ipynb/primary-processing/11-generate_per_bodysite_results.md]: Ah.  I see that now.  Cool.

**Comment by JWDebelius** [comment]
Since all comments have been addressed and tests are passing, could this be merged?

Thanks.

**Comment by wasade** [comment]
Awesome!!
